server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -367,8 +366,7 @@ public ClusterState addIndexTemplateV2(final ClusterState currentState, final bo
                     .map(e -> e.getKey() + " => " + e.getValue())
                     .collect(Collectors.joining(",")),
                 name);
-            logger.warn(warning);
-            deprecationLogger.deprecatedAndMaybeLog("index_template_pattern_overlap", warning);
+            warningLogger.logAndAddWarning(warning);
```

Comment:
this is the usage of the refactoring from this PR.
this log line always log  - to server - and adds a warning. 
The type might be a bit confusing. Maybe this should 2 lines:
1. to log to a server file - regular logger
2. add a warning to a headers
@jakelandis wdyt?

Reply:
++ for 2 lines.  We should also probably make `ThrottlingAndHeaderWarningLogger` package private.